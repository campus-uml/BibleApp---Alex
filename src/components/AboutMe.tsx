import {
  Card,
  CardContent,
  CardHeader,

} from "@/components/ui/card";
import { Avatar, AvatarFallback, AvatarImage } from "@/components/ui/avatar";
import { Badge } from "@/components/ui/badge";
import { FacebookIcon, GithubIcon, LinkedinIcon } from "lucide-react";

export const AboutMe = () => {
  return (
    <Card className="w-full max-w-md mx-auto ">
      <CardHeader className="text-center">
      </CardHeader>
      <CardContent className="flex flex-col items-center space-y-4">
        <Avatar className="w-24 h-24">
          <AvatarImage src="/avatar.jpg" alt="Foto del desarrollador" />
          <AvatarFallback>
            <img
              src="https://i.pinimg.com/736x/3e/f9/5e/3ef95eb2be325c976ac5e33d3efb1224.jpg"
              alt="Perfil"
            />
          </AvatarFallback>
        </Avatar>
        <div className="text-center">
          <h2 className="text-xl font-semibold">Eddy Alexis Talavera</h2>
          <p className="text-sm text-muted-foreground">Desarrollador junior</p>
        </div>
        <div className="flex space-x-2">
          <Badge variant="secondary">React</Badge>
          <Badge variant="secondary">TypeScript</Badge>
          <Badge variant="secondary">Tailwind CSS</Badge>
        </div>
        <p className="text-sm text-center max-w-xs">
          Apasionado por crear aplicaciones web intuitivas y eficientes.
          Especializado en React y TypeScript.
        </p>
        <div className="flex space-x-4">
          <a
            href="https://github.com/Alex200207"
            target="_blank"
            rel="noopener noreferrer"
            className="text-muted-foreground hover:text-primary"
          >
            <GithubIcon className="w-6 h-6" />
          </a>
          <a
            href="https://www.linkedin.com/in/alex-talavera-57a8b12a2/"
            target="_blank"
            rel="noopener noreferrer"
            className="text-muted-foreground hover:text-primary"
          >
            <LinkedinIcon className="w-6 h-6" />
          </a>
          <a
            href="https://www.facebook.com/alexis.talavera.503?locale=es_LA"
            target="_blank"
            rel="noopener noreferrer"
            className="text-muted-foreground hover:text-primary"
          >
            <FacebookIcon className="w-6 h-6" />
          </a>
        </div>
      </CardContent>
    </Card>
  );
};
