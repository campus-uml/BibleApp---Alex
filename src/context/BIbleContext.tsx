import React, {
  createContext,
  useContext,
  useEffect,
  useRef,
  useState,
} from "react";
import {
  getBibles,
  getChapters,
  loadChapterVersesFromAPI,
  searchBible,
} from "../services/getData";
import { BibleBooks, Chapter, Verse } from "../types/index";
import { useSidebar } from "@/components/ui/sidebar";

const formatVerseText = (text: string) => {
  return text.toLowerCase().replace(/^([a-z])/, (match) => match.toUpperCase());
};

const isTitleOrHeading = (text: string) => {
  const titleKeywords = ["Capítulo", "Libro", "Título"];
  return titleKeywords.some((keyword) => text.includes(keyword));
};

interface BibleContextProps {
  bibleVerse: BibleBooks[];
  selectedBook: BibleBooks["id"];
  bibleVerseChapters: Chapter[];
  chapterVerses: Verse[];
  scrollAreaRef: React.RefObject<HTMLDivElement>;
  handleBook: (bookId: string) => void;
  loadChapterVerses: (chapterId: string) => void;
  searchBibleVerse: (query: string) => void;
  toggleSidebar: () => void;
  
}

const BibleContext = createContext<BibleContextProps | undefined>(undefined);

interface BibleProviderProps {
  children: React.ReactNode;
}

export const BibleProvider: React.FC<BibleProviderProps> = ({ children }) => {
  const [bibleVerse, setBibleVerse] = useState<BibleBooks[]>([]);
  const [selectedBook, setSelectedBook] = useState<BibleBooks["id"]>("GEN");
  const [bibleVerseChapters, setBibleVerseChapters] = useState<Chapter[]>([]);
  const [chapterVerses, setChapterVerses] = useState<Verse[]>([]);
  const scrollAreaRef = useRef<HTMLDivElement>(null);
  const { toggleSidebar } = useSidebar();

  useEffect(() => {
    fetchData();
  }, []);

  useEffect(() => {
    if (selectedBook) {
      loadChapters(selectedBook);
    }
  }, [selectedBook]);

  useEffect(() => {
    if (bibleVerseChapters.length > 0) {
      loadChapterVerses(bibleVerseChapters[0].id);
    }
  }, [bibleVerseChapters]);

  const loadChapterVerses = async (chapterId: string) => {
    try {
      const response = await loadChapterVersesFromAPI(chapterId);
      if (!response.data) {
        console.error(
          "Error: No se encontró contenido en los versículos",
          response
        );
        return;
      }

      const content = response.data.content.replace(/<\/?[^>]+(>|$)/g, "");
      const versePattern = /(\d+)([^0-9]+)/g;
      const versesArray: Verse[] = [];
      let match;

      while ((match = versePattern.exec(content)) !== null) {
        const verseNumber = match[1];
        const verseText = formatVerseText(match[2].trim());

        if (!isTitleOrHeading(verseText)) {
          versesArray.push({
            id: `${chapterId}-${verseNumber}`,
            reference: `Versículo ${verseNumber}`,
            text: verseText,
          });
        }
      }

      setChapterVerses(versesArray);
    } catch (error) {
      console.error("Error fetching chapter content:", error);
    }
  };

  const fetchData = async () => {
    try {
      const response = await getBibles();
      if (response && response.data) {
        setBibleVerse(response.data);
      } else {
        console.error("Unexpected API response structure:", response);
      }
    } catch (error) {
      console.error("Error fetching bibles:", error);
    }
  };

  const handleBook = (bookId: string) => {
    setSelectedBook(bookId);
    toggleSidebar();
  };

  const loadChapters = async (bookId: string) => {
    try {
      const response = await getChapters(bookId);
      if (response && response.data) {
        setBibleVerseChapters(response.data);
      } else {
        console.error("Unexpected API response structure:", response);
      }
    } catch (error) {
      console.error("Error fetching chapters:", error);
    }
  };

  const searchBibleVerse = async (query: string) => {
    try {
      const response = await searchBible(query);
      if (response && response.data) {
        setBibleVerse(response.data);
      } else {
        console.error("Unexpected API response structure:", response);
      }
    } catch (error) {
      console.error("Error fetching bible verse:", error);
    }
  };
  return (
    <BibleContext.Provider
      value={{
        bibleVerse,
        selectedBook,
        bibleVerseChapters,
        chapterVerses,
        handleBook,
        loadChapterVerses,
        scrollAreaRef,
        searchBibleVerse,
        toggleSidebar,
      }}
    >
      {children}
    </BibleContext.Provider>
  );
};

// eslint-disable-next-line react-refresh/only-export-components
export const useBible = (): BibleContextProps => {
  const context = useContext(BibleContext);
  if (!context) {
    throw new Error("useBible must be used within a BibleProvider");
  }
  return context;
};
