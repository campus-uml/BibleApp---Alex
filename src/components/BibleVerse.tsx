import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
import { ScrollArea, ScrollBar } from "@/components/ui/scroll-area";
import { useBible } from "../context/BIbleContext";
import { Copy, Heart, ArrowLeft, Share, Play, Book } from "lucide-react";
import type { SearchResults, Verse } from "@/types";
import { Button } from "@/components/ui/button";
import { useActions } from "@/Hooks/useActions";

interface BibleVerseProps {
  searchResults: SearchResults | null;
  onClearSearch: () => void;
  addFavorite: (verse: Verse) => void;
  removeFavorite: (verseId: string) => void;
  favorites: Verse[];
}

export const BibleVerse = ({
  searchResults,
  onClearSearch,
  addFavorite,
  favorites,
  removeFavorite,
}: BibleVerseProps) => {
  const {
    bibleVerseChapters,
    chapterVerses,
    loadChapterVerses,
    scrollAreaRef,
    selectedBook,
    bibleVerse,
  } = useBible();
  const { copyToClipboard, shareToWhatsApp } = useActions();

  const bookName = (selectedBook: string) => {
    const book = bibleVerse.find((book) => book.id === selectedBook);
    return book ? book.name : "";
  };

  return (
    <div className="min-h-screen bg-gray-50">
      <Tabs
        defaultValue="1"
        className="w-full max-w-[21.2rem] md:max-w-6xl mx-auto"
      >
        <div className="sticky top-0 bg-gray-50 z-10 pb-4">
          {searchResults ? (
            <div className="flex items-center justify-between mb-4">
              <Button
                variant="outline"
                onClick={onClearSearch}
                className="flex items-center gap-2"
              >
                <ArrowLeft size={16} />
                Volver
              </Button>
              <h2 className="text-xl font-semibold text-gray-800">
                Resultados de búsqueda
              </h2>
            </div>
          ) : (
            <>
              <h1 className="text-2xl font-bold text-center text-gray-800 mb-4">
                {bookName(selectedBook)}
              </h1>
              {bibleVerseChapters.length > 0 ? (
                <TabsList className="h-12 items-center justify-start rounded-md bg-muted p-1 text-muted-foreground w-full overflow-hidden">
                  <ScrollArea
                    ref={scrollAreaRef}
                    className="w-full overflow-x-auto whitespace-nowrap scrollbar-thin scrollbar-thumb-gray-300 scrollbar-track-gray-100"
                  >
                    <div className="flex items-center gap-2 py-1">
                      {bibleVerseChapters.map((chapter) => (
                        <TabsTrigger
                          key={chapter.id}
                          value={chapter.number.toString()}
                          onClick={() => loadChapterVerses(chapter.id)}
                          className="ring-offset-background focus-visible:ring-ring data-[state=active]:bg-background data-[state=active]:text-foreground px-3 py-1.5 rounded-md"
                        >
                          {chapter.number}
                        </TabsTrigger>
                      ))}
                    </div>
                    <ScrollBar orientation="horizontal" />
                  </ScrollArea>
                </TabsList>
              ) : (
                <div className="text-center py-8">
                  <Book className="w-12 h-12 text-slate-300 mx-auto mb-3" />
                  <p className="text-lg font-semibold text-slate-600">
                    No hay versículos aún
                  </p>
                  <p className="text-slate-500 mb-4">
                    Selecciona en la parte superior para comenzar
                  </p>
                </div>
              )}
            </>
          )}
        </div>

        {searchResults &&
        searchResults.verses &&
        searchResults.verses.length > 0 ? (
          <div className="space-y-4 mt-4">
            {searchResults.verses.map((verse, index) => (
              <div key={index} className="bg-white shadow-md rounded-lg p-2">
                <h3 className="text-sm text-slate-500 font-semibold mb-2">
                  {verse.id}
                </h3>
                <p className="text-base text-gray-700">{verse.text}</p>
                <div className="flex justify-end gap-4 mt-4">
                  <button
                    onClick={() =>
                      favorites.some((fav) => fav.id === verse.id)
                        ? removeFavorite(verse.id)
                        : addFavorite({ ...verse, reference: verse.id })
                    }
                    className="mt-2 focus:outline-none"
                    aria-label={
                      favorites.some((fav) => fav.id === verse.id)
                        ? "remover"
                        : "agregar"
                    }
                  >
                    <Heart
                      className={`w-6 h-6 ${
                        favorites.some((fav) => fav.id === verse.id)
                          ? "text-red-500"
                          : "text-gray-900"
                      }`}
                    />
                  </button>
                  <button className="text-gray-600 hover:text-blue-500 transition-colors">
                    <Play size={20} />
                  </button>
                  <button
                    onClick={() => copyToClipboard(verse.text)}
                    className="text-gray-600 hover:text-green-500 transition-colors"
                  >
                    <Copy size={20} />
                  </button>
                  <button
                    onClick={() => shareToWhatsApp(verse.text)}
                    className="text-gray-600 hover:text-purple-500 transition-colors"
                  >
                    <Share size={20} />
                  </button>
                </div>
              </div>
            ))}
          </div>
        ) : (
          <div className="space-y-4 mt-4">
            {bibleVerseChapters.map((chapter) => (
              <TabsContent
                key={chapter.id}
                value={chapter.number.toString()}
                className="bg-white rounded-lg shadow-sm p-2"
              >
                <h2 className="text-xl text-center font-semibold mb-4 text-gray-800">
                  Capítulo {chapter.number}
                </h2>
                <div className="space-y-4">
                  {chapterVerses.map((verse, index) => (
                    <div
                      key={index}
                      className="bg-white shadow-md rounded-lg p-4"
                    >
                      <h3 className="text-sm text-slate-500 font-semibold mb-2">
                        {verse.reference}
                      </h3>
                      <p className="text-gray-700 leading-relaxed italic">
                        {verse.text}
                      </p>
                      <div className="flex justify-end gap-4 mt-4">
                        <button
                          onClick={() =>
                            favorites.some((fav) => fav.id === verse.id)
                              ? removeFavorite(verse.id)
                              : addFavorite({ ...verse, reference: verse.id })
                          }
                          className="mt-2 focus:outline-none"
                          aria-label={
                            favorites.some((fav) => fav.id === verse.id)
                              ? "remover"
                              : "agregar"
                          }
                        >
                          <Heart
                            size={20}
                            className={` ${
                              favorites.some((fav) => fav.id === verse.id)
                                ? "text-red-500"
                                : "text-gray-900"
                            }`}
                          />
                        </button>
                        <button className="text-gray-600 hover:text-blue-500 transition-colors">
                          <Play size={20} />
                        </button>
                        <button
                          onClick={() => copyToClipboard(verse.text)}
                          className="text-gray-600 hover:text-green-500 transition-colors"
                        >
                          <Copy size={20} />
                        </button>
                        <button
                          onClick={() => shareToWhatsApp(verse.text)}
                          className="text-gray-600 hover:text-purple-500 transition-colors"
                        >
                          <Share size={20} />
                        </button>
                      </div>
                    </div>
                  ))}
                </div>
              </TabsContent>
            ))}
          </div>
        )}
      </Tabs>
    </div>
  );
};

export default BibleVerse;
