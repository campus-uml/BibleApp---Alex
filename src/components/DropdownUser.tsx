import { Button } from "@/components/ui/button";
import {
  DropdownMenu,
  DropdownMenuContent,
  DropdownMenuItem,
  DropdownMenuLabel,
  DropdownMenuSeparator,
  DropdownMenuTrigger,
} from "@/components/ui/dropdown-menu";
import { Avatar, AvatarFallback, AvatarImage } from "@/components/ui/avatar";
import LogoutButton from "@/components/login/LogoutButton";
import { Settings, UserIcon } from "lucide-react";
import { useAuth } from "@/context/AuthContext";

interface DropdownUserProps {
  avatarUrl: string | null;
}

export function DropdownUser({ avatarUrl }: DropdownUserProps) {
  const { user } = useAuth();

  const displayName = user?.user_metadata.full_name || user?.user_metadata.name;

  console.log("user", user);
  return (
    <DropdownMenu>
      <DropdownMenuTrigger asChild>
        <Button
          variant="ghost"
          size="icon"
          className="rounded-full hover:bg-accent"
        >
          <Avatar className="h-8 w-8 transition-transform hover:scale-110">
            <AvatarImage src={avatarUrl ?? undefined} alt="User Avatar" />
            <AvatarFallback className="bg-primary text-primary-foreground">
            {user?.email?.[0].toUpperCase() ?? "?"}
            </AvatarFallback>
          </Avatar>
        </Button>
      </DropdownMenuTrigger>
      <DropdownMenuContent align="end" className="w-56">
        <div className="flex items-center justify-start gap-2 p-2">
          <Avatar className="h-8 w-8">
            <AvatarImage src={avatarUrl ?? undefined} alt="User Avatar" />
            <AvatarFallback className="bg-primary text-primary-foreground">
            {user?.email?.[0].toUpperCase() ?? "?"}
            </AvatarFallback>
          </Avatar>
          <div className="flex flex-col space-y-1">
            <DropdownMenuLabel className="font-medium">
              {displayName}
            </DropdownMenuLabel>
            <p className="text-xs text-muted-foreground">
              @{user?.user_metadata.user_name || user?.email}
            </p>
          </div>
        </div>
        <DropdownMenuSeparator />
        <DropdownMenuItem>
          <UserIcon className="mr-2 h-4 w-4" />
          <span>Perfil</span>
        </DropdownMenuItem>
        <DropdownMenuItem>
          <Settings className="mr-2 h-4 w-4" />
          <span>Configuraci√≥n</span>
        </DropdownMenuItem>
        <DropdownMenuSeparator />
        <DropdownMenuItem asChild>
          <LogoutButton />
        </DropdownMenuItem>
      </DropdownMenuContent>
    </DropdownMenu>
  );
}
