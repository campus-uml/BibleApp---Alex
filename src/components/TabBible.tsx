import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
import { useBible } from "../context/BIbleContext";
import { Button } from "./ui/button";
import { ScrollArea } from "./ui/scroll-area";
import { newTestamentIds, oldTestamentIds } from "../utils/Abreviations";

export const TabBible = () => {
  const { bibleVerse, handleBook } = useBible();

  const oldTestamentBooks = bibleVerse.filter((book) =>
    oldTestamentIds.includes(book.id)
  );
  const newTestamentBooks = bibleVerse.filter((book) =>
    newTestamentIds.includes(book.id)
  );

  const renderBookList = (books: typeof oldTestamentBooks) => (
    <ScrollArea className="h-[80vh] w-full rounded-md border">
      <div className="p-4">
        {books.length > 0 ? (
          books.map((v, index) => (
            <Button
              key={index}
              onClick={() => handleBook(v.id)}
              variant="ghost"
              className="w-full justify-start text-left font-normal mb-2 hover:bg-accent hover:text-accent-foreground"
            >
              <span className="text-base">{v.name}</span>
            </Button>
          ))
        ) : (
          <p className="text-muted-foreground">No data available</p>
        )}
      </div>
    </ScrollArea>
  );

  return (
    <Tabs defaultValue="Antiguo" className="w-full">
      <TabsList className="grid w-full grid-cols-3 mb-4">
        <TabsTrigger value="Antiguo">Antiguo</TabsTrigger>
        <TabsTrigger value="Nuevo">Nuevo</TabsTrigger>
        <TabsTrigger value="Frecuentes">Frecuentes</TabsTrigger>
      </TabsList>
      <TabsContent value="Antiguo">
        {renderBookList(oldTestamentBooks)}
      </TabsContent>
      <TabsContent value="Nuevo">
        {renderBookList(newTestamentBooks)}
      </TabsContent>
      <TabsContent value="Frecuentes">
        <div className="rounded-md border p-4">
          <h3 className="text-lg font-medium mb-2">Información frecuente</h3>
          <p className="text-muted-foreground">
            Aquí puedes agregar información sobre los libros más frecuentemente
            consultados o versículos populares.  (en desarrollo)
          </p>
        </div>
      </TabsContent>
    </Tabs>
  );
};
