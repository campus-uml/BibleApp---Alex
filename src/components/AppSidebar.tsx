import * as React from "react";
import { GalleryVerticalEnd, Search } from "lucide-react";
import {
  Sidebar,
  SidebarContent,
  SidebarGroup,
  SidebarHeader,
  SidebarMenu,
  SidebarMenuButton,
} from "@/components/ui/sidebar";
import { TabBible } from "./TabBible";
import { Button } from "@/components/ui/button";
import {
  Tooltip,
  TooltipProvider,
  TooltipTrigger,
} from "@/components/ui/tooltip";
import { DropdownUser } from "./DropdownUser";
import { useState } from "react";
import { useAuth } from "@/context/AuthContext";
import SearchBook from "./SearchBook";

export function AppSidebar({ ...props }: React.ComponentProps<typeof Sidebar>) {
  const [showSearch, setShowSearch] = useState(false);
  const [searchTerm, setSearchTerm] = useState("");
  const { avatarUrl, user } = useAuth();

  const handleSearch = (term: string) => {
    setSearchTerm(term);
  };

  return (
    <Sidebar {...props}>
      <SidebarHeader className="flex flex-col space-y-2 p-2">
        <div className="flex items-center justify-between">
          <SidebarMenuButton size="sm" asChild>
            <a href="#" className="flex items-center space-x-2">
              <div className="flex aspect-square size-8 items-center justify-center rounded-lg bg-sidebar-primary text-sidebar-primary-foreground">
                <GalleryVerticalEnd className="size-4" />
              </div>
              <span className="font-semibold">BibleApp</span>
            </a>
          </SidebarMenuButton>
          <div className="flex items-center space-x-2">
            <TooltipProvider>
              <Tooltip>
                <TooltipTrigger asChild>
                  <Button
                    variant="ghost"
                    size="icon"
                    onClick={() => setShowSearch(!showSearch)}
                  >
                    <Search className="h-4 w-4" />
                  </Button>
                </TooltipTrigger>
              </Tooltip>
            </TooltipProvider>
            <DropdownUser avatarUrl={avatarUrl} user={user} />
          </div>
        </div>
        {showSearch && <SearchBook onSearch={handleSearch} />}
      </SidebarHeader>
      <SidebarContent>
        <SidebarGroup>
          <SidebarMenu>
            <TabBible searchTerm={searchTerm} />
          </SidebarMenu>
        </SidebarGroup>
      </SidebarContent>
    </Sidebar>
  );
}
