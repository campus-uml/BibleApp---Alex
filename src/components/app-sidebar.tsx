import * as React from "react";
import { GalleryVerticalEnd, Search } from "lucide-react";
import {
  Sidebar,
  SidebarContent,
  SidebarGroup,
  SidebarHeader,
  SidebarMenu,
  SidebarMenuButton,
} from "@/components/ui/sidebar";
import { TabBible } from "./TabBible";
import { SearchInput } from "../components/searchInput";
import { Button } from "@/components/ui/button";
import {
  Tooltip,
  TooltipContent,
  TooltipProvider,
  TooltipTrigger,
} from "@/components/ui/tooltip";

export function AppSidebar({ ...props }: React.ComponentProps<typeof Sidebar>) {
  const [showSearch, setShowSearch] = React.useState(false);

  return (
    <Sidebar {...props}>
      <SidebarHeader className="flex flex-col space-y-2 p-2">
        <div className="flex items-center justify-between">
          <SidebarMenuButton size="sm" asChild>
            <a href="#" className="flex items-center space-x-2">
              <div className="flex aspect-square size-8 items-center justify-center rounded-lg bg-sidebar-primary text-sidebar-primary-foreground">
                <GalleryVerticalEnd className="size-4" />
              </div>
              <span className="font-semibold">BibleApp</span>
            </a>
          </SidebarMenuButton>
          <TooltipProvider>
            <Tooltip>
              <TooltipTrigger asChild>
                <Button
                  variant="ghost"
                  size="icon"
                  onClick={() => setShowSearch(!showSearch)}
                >
                  <Search className="h-4 w-4" />
                </Button>
              </TooltipTrigger>
              <TooltipContent>
                <p>Buscar en la Biblia</p>
              </TooltipContent>
            </Tooltip>
          </TooltipProvider>
        </div>
        {showSearch && <SearchInput />}
      </SidebarHeader>
      <SidebarContent>
        <SidebarGroup>
          <SidebarMenu>
            <TabBible />
          </SidebarMenu>
        </SidebarGroup>
      </SidebarContent>
    </Sidebar>
  );
}
