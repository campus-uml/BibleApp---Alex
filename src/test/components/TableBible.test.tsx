import { render, screen } from "@testing-library/react";
import { TabBible } from "../../components/TabBible";
import { useBible } from "../../context/BIbleContext";
import { vi, describe, beforeEach, test, expect, Mock } from "vitest";

vi.mock("../../context/BIbleContext", () => ({
  useBible: vi.fn(),
}));

describe("TabBible Component", () => {
  const mockHandleBook = vi.fn();
  const mockBibleVerse = [
    { id: "gen", name: "Génesis" },
    { id: "exo", name: "Éxodo" },
  ];

  beforeEach(() => {
    (useBible as Mock).mockReturnValue({
      bibleVerse: mockBibleVerse,
      handleBook: mockHandleBook,
    });
  });

test(" deberia renderizar pestañas y lista de libros", () => {
    render(<TabBible searchTerm="" />);
    expect(screen.getByText("Antiguo")).toBeInTheDocument();
    expect(screen.getByText("Nuevo")).toBeInTheDocument();
    expect(screen.getByText("Frecuentes")).toBeInTheDocument();
});

test("filtra libros basado en el término de búsqueda", () => {
    render(<TabBible searchTerm="g" />);
    expect(screen.queryByText("Génesis")).not.toBeInTheDocument();
});
});
