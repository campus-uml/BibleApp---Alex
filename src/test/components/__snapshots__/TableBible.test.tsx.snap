// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TabBible Component > deberia renderizar pestañas y lista de libros 1`] = `
<div>
  <div
    class="w-full"
    data-orientation="horizontal"
    dir="ltr"
  >
    <div
      aria-orientation="horizontal"
      class="h-9 items-center justify-center rounded-lg bg-muted p-1 text-muted-foreground grid w-full grid-cols-3 mb-4"
      data-orientation="horizontal"
      role="tablist"
      style="outline: none;"
      tabindex="0"
    >
      <button
        aria-controls="radix-:r0:-content-Antiguo"
        aria-selected="true"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-background data-[state=active]:text-foreground data-[state=active]:shadow"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="active"
        id="radix-:r0:-trigger-Antiguo"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Antiguo
      </button>
      <button
        aria-controls="radix-:r0:-content-Nuevo"
        aria-selected="false"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-background data-[state=active]:text-foreground data-[state=active]:shadow"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="inactive"
        id="radix-:r0:-trigger-Nuevo"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Nuevo
      </button>
      <button
        aria-controls="radix-:r0:-content-Frecuentes"
        aria-selected="false"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-background data-[state=active]:text-foreground data-[state=active]:shadow"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="inactive"
        id="radix-:r0:-trigger-Frecuentes"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Frecuentes
      </button>
    </div>
    <div
      aria-labelledby="radix-:r0:-trigger-Antiguo"
      class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
      data-orientation="horizontal"
      data-state="active"
      id="radix-:r0:-content-Antiguo"
      role="tabpanel"
      style="animation-duration: 0s;"
      tabindex="0"
    >
      <div
        class="relative overflow-hidden h-[80vh] w-full rounded-md border"
        dir="ltr"
        style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
      >
        <style>
          [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
        </style>
        <div
          class="h-full w-full rounded-[inherit]"
          data-radix-scroll-area-viewport=""
          style="overflow-x: hidden; overflow-y: scroll;"
        >
          <div
            style="min-width: 100%; display: table;"
          >
            <div
              class="p-4"
            >
              <p
                class="text-muted-foreground"
              >
                No se encontraron libros
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="radix-:r0:-trigger-Nuevo"
      class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
      data-orientation="horizontal"
      data-state="inactive"
      hidden=""
      id="radix-:r0:-content-Nuevo"
      role="tabpanel"
      tabindex="0"
    />
    <div
      aria-labelledby="radix-:r0:-trigger-Frecuentes"
      class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
      data-orientation="horizontal"
      data-state="inactive"
      hidden=""
      id="radix-:r0:-content-Frecuentes"
      role="tabpanel"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`TabBible Component > filtra libros basado en el término de búsqueda 1`] = `
<div>
  <div
    class="w-full"
    data-orientation="horizontal"
    dir="ltr"
  >
    <div
      aria-orientation="horizontal"
      class="h-9 items-center justify-center rounded-lg bg-muted p-1 text-muted-foreground grid w-full grid-cols-3 mb-4"
      data-orientation="horizontal"
      role="tablist"
      style="outline: none;"
      tabindex="0"
    >
      <button
        aria-controls="radix-:r4:-content-Antiguo"
        aria-selected="true"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-background data-[state=active]:text-foreground data-[state=active]:shadow"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="active"
        id="radix-:r4:-trigger-Antiguo"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Antiguo
      </button>
      <button
        aria-controls="radix-:r4:-content-Nuevo"
        aria-selected="false"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-background data-[state=active]:text-foreground data-[state=active]:shadow"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="inactive"
        id="radix-:r4:-trigger-Nuevo"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Nuevo
      </button>
      <button
        aria-controls="radix-:r4:-content-Frecuentes"
        aria-selected="false"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-background data-[state=active]:text-foreground data-[state=active]:shadow"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="inactive"
        id="radix-:r4:-trigger-Frecuentes"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Frecuentes
      </button>
    </div>
    <div
      aria-labelledby="radix-:r4:-trigger-Antiguo"
      class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
      data-orientation="horizontal"
      data-state="active"
      id="radix-:r4:-content-Antiguo"
      role="tabpanel"
      style="animation-duration: 0s;"
      tabindex="0"
    >
      <div
        class="relative overflow-hidden h-[80vh] w-full rounded-md border"
        dir="ltr"
        style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
      >
        <style>
          [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
        </style>
        <div
          class="h-full w-full rounded-[inherit]"
          data-radix-scroll-area-viewport=""
          style="overflow-x: hidden; overflow-y: scroll;"
        >
          <div
            style="min-width: 100%; display: table;"
          >
            <div
              class="p-4"
            >
              <p
                class="text-muted-foreground"
              >
                No se encontraron libros
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="radix-:r4:-trigger-Nuevo"
      class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
      data-orientation="horizontal"
      data-state="inactive"
      hidden=""
      id="radix-:r4:-content-Nuevo"
      role="tabpanel"
      tabindex="0"
    />
    <div
      aria-labelledby="radix-:r4:-trigger-Frecuentes"
      class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
      data-orientation="horizontal"
      data-state="inactive"
      hidden=""
      id="radix-:r4:-content-Frecuentes"
      role="tabpanel"
      tabindex="0"
    />
  </div>
</div>
`;
